spring-cloud-kubernetes-leader/src/main/java/org/springframework/cloud/kubernetes/leader/LeaderRecordWatcher.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2013-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.springframework.cloud.kubernetes.leader;
+
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClientException;
+import io.fabric8.kubernetes.client.Watch;
+import io.fabric8.kubernetes.client.Watcher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Gytis Trikleris
+ */
+public class LeaderRecordWatcher implements Watcher<ConfigMap> {
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(LeaderRecordWatcher.class);
+
+	private final LeadershipController leadershipController;
+
+	private final LeaderProperties leaderProperties;
+
+	private final KubernetesClient kubernetesClient;
+
+	private Watch watch;
```

Comment:
i think that this is only written by `start` and `onClose`. With reasonable use we should be fine regarding thread safety. Still it would be nice to either comment or guard.

Reply:
yeah. thats what i meant as guard.

Reply:
@iocanel I've added the locks around the status update blocks in watchers